providers/gitlab.go

```diff
@@ -34,28 +39,89 @@ func NewGitLabProvider(p *ProviderData) *GitLabProvider {
 		p.ValidateURL = &url.URL{
 			Scheme: "https",
 			Host:   "gitlab.com",
-			Path:   "/api/v4/user",
+			Path:   "/api/v4",
 		}
 	}
-	if p.Scope == "" {
-		p.Scope = "read_user"
-	}
+	p.Scope += " read_user"
 	return &GitLabProvider{ProviderData: p}
 }
 
 // GetEmailAddress returns the Account email address
 func (p *GitLabProvider) GetEmailAddress(s *SessionState) (string, error) {
+	if p.Group != "" {
+		groupsJSON, groupsJSONError := data(p.ValidateURL.String()+"/groups?access_token="+s.AccessToken, s)
```

Comment:
I'm not a fan of the `groupsJSONError` name, any reason not to use the idiomatic `err` throughout this method?
Really long error names distract from the code as you are reading it, `err` is easily recognisable and filterable as you are reading IMO

Reply:
It is a coding style I use a lot in other projects. I made the experience, that using just `err` for every error leads to confusion, when it comes to several error handling blocks. So I started to use more speaking variable names in go code, which helped a lot.